src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Finder\Iterator;
+
+use Symfony\Component\Finder\Gitignore;
+
+final class VcsIgnoredFilterIterator extends \FilterIterator
+{
+    /**
+     * @var string
+     */
+    private $baseDir;
+
+    /**
+     * @var array<string, string|null>
+     */
+    private $gitignoreFilesCache = [];
+
+    public function __construct(\Iterator $iterator, string $baseDir)
+    {
+        $this->baseDir = $baseDir;
+
+        parent::__construct($iterator);
+    }
+
+    public function accept(): bool
+    {
+        $file = $this->current();
+
+        $fileRealPath = $file->getRealPath();
```

Comment:
@julienfalque realpath can be slow and I do not think this is actually correct by git spec at all

Imagine a symlink excluded by a .gitignore. The symlink will not be ignored a it is resolved here.

see https://stackoverflow.com/questions/40460216/how-do-i-make-git-ignore-symlink#40461160